tests/integration/main.rs

```diff
@@ -601,3 +613,271 @@ async fn test_transaction_with_update() -> Result<(), DatastorersError> {
 
     Ok(())
 }
+
+
+
+#[tokio::test]
+#[cfg_attr(not(feature = "integration_tests"), ignore)]
+async fn test_query_by_props() -> Result<(), DatastorersError> {
+    let connection = create_test_connection().await;
+
+    // Save 3 entities, 2 with the same name
+    let expected_result_entity = generate_random_entity().commit(&connection).await?;
+    let mut duplicated_entity = generate_random_entity();
+    // For those with duplicated names we add different numeric props in order
+    // to be able to test queries with multiple conditions
+    duplicated_entity.prop_int = generate_random_int();
+    let _duplicated_one = duplicated_entity.clone().commit(&connection).await?;
+    duplicated_entity.prop_int = generate_random_int();
+    let duplicated_two = duplicated_entity.clone().commit(&connection).await?;
+
+    assert_ne!(expected_result_entity.prop_int, duplicated_entity.prop_int);
+
+    let fetched_entity = TestEntity::query()
+        // TODO - here we have the datastore property name - that is doable, but no good
+        // Suggestion - use "prop_string" here and then implement a mapper method on the struct
+        // That method can be part of the DatastorersQueryable trait
+        .filter(String::from("Name"), Operator::Equal, expected_result_entity.prop_string)?
```

Comment:
this we need to descide, shall it be the datastore name (like here), or the property name from the struct, something like:

` .filter(String::from("prop_string"), Operator::Equal, expected_result_entity.prop_string)?`

Would be quite easy to change to the struct property name, and I think that would make more sense, or?